### PR TITLE
Update to SDL2 

### PR DESCRIPTION
**Pull request by selaux**
Issue: #206

### CODE REVIEW

File: sgp/Input.cc [hunk 1]
```diff
@@ -184,11 +190,23 @@ void MouseButtonUp(const SDL_MouseButtonEvent* BtnEv)
 	}
 }
 
+void MouseWheelScroll(const SDL_MouseWheelEvent* WheelEv) {
```
Comment (SergeL) [inline sgp/Input.cc]: :nail_care: New line before `{` missing

File: sgp/Video.cc [hunk 2]
```diff
@@ -142,27 +124,64 @@ void InitializeVideoManager(void)
...
+	           SCREEN_HEIGHT,
+	           PIXEL_DEPTH,
+               Rmask,
+               Gmask,
+               Bmask,
+               Amask
+    );
+
+    if (ScreenBuffer == NULL) {
+        SLOGD("Topic", "SDL_CreateRGBSurface for ScreenBuffer failed: %s\n", SDL_GetError())
```
Comment (derSteFfi) [inline sgp/Video.cc]: i see that a ';' is not neccessary here, but every other place in the codebase puts semicolons there.
The #define for SLOGD adds it, but i would prefer consistency across the code

Comment (selaux) [inline sgp/Video.cc]: :+1: the topic seems wrong to me as well...

File: sgp/Video.cc [hunk 3]
```diff
@@ -136,8 +136,9 @@ void InitializeVideoManager(void)
                               g_window_flags);
 	GameRenderer = SDL_CreateRenderer(g_game_window, -1, 0);
 	SDL_RenderSetLogicalSize(GameRenderer, SCREEN_WIDTH, SCREEN_HEIGHT);
+  SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "linear");
 
-		SDL_Surface* windowIcon = SDL_CreateRGBSurfaceFrom(
+  SDL_Surface* windowIcon = SDL_CreateRGBSurfaceFrom(
```
Comment (derSteFfi) [inline sgp/Video.cc]: this looks like you introduced spaces here instead of tabs...

**Comment by selaux** [comment]
Allright now I'm able at least to build this branch on every OS. Currently I'm working on how the SDL2 framework needs to be distributed for OS X (that seems to have changed)

**Comment by selaux** [comment]
Finally everything seems to be working now. Please test. Attached are the releases I built with the SDL2 dependencies. I split the update of the dependencies to a separate PR, so actual changes that needed to be done are more emphasized here (otherwise GitHub wouldn't render the full diff).

[sdl2-releases.zip](https://github.com/ja2-stracciatella/ja2-stracciatella/files/454044/sdl2-releases.zip)

**Comment by misanthropos** [comment]
I just tried the amd64 version on Ubuntu. It seems libsdl2 has a different name but still got installed and I could start it. However I cant continue the SaveGame from before the lawfabric.

error [queen command] assertion failed in campaign_types.h line 11

has anybody checked if one can play through a sector?

**Comment by lynxlynxlynx** [comment]
That would mean an invalid sector number, but it's likely to just be the aggressive asserts thing we have now in master (they're not ifdefed out anymore). So the question is, do you have the same problem with sdl1 or not?

**Comment by selaux** [comment]
We really need either to playtest a lot for the asserts, or disable them before the next release 😉

**Comment by lynxlynxlynx** [comment]
I'd just disable them for non-debug builds. Too many permutations for it to be live for the masses.

**Comment by lynxlynxlynx** [comment]
I've built it and tested a bit all around — seems to work fine. :)

**Comment by lynxlynxlynx** [comment]
Yay! Good job. :)

**Comment by derSteFfi** [comment]
Agree! Great one :)

**Pull request merged**